x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/index_view_logic.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { i18n } from '@kbn/i18n';
+
+import { SyncStatus } from '../../../../../common/types/connectors';
+import { Actions } from '../../../shared/api_logic/create_api_logic';
+import {
+  flashAPIErrors,
+  clearFlashMessages,
+  flashSuccessToast,
+} from '../../../shared/flash_messages';
+import { StartSyncApiLogic, StartSyncArgs } from '../../api/connector_package/start_sync_api_logic';
+import {
+  FetchIndexApiLogic,
+  FetchIndexApiParams,
+  FetchIndexApiResponse,
+} from '../../api/index/fetch_index_api_logic';
+import { ElasticsearchViewIndex, IngestionMethod, IngestionStatus } from '../../types';
+import {
+  getIngestionMethod,
+  getIngestionStatus,
+  getLastUpdated,
+  indexToViewIndex,
+} from '../../utils/indices';
+
+export type IndicesActions = Pick<
+  Actions<StartSyncArgs, {}>,
+  'makeRequest' | 'apiSuccess' | 'apiError'
+> & {
+  fetchIndexSuccess: Actions<FetchIndexApiParams, FetchIndexApiResponse>['apiSuccess'];
+  startSync(): void;
+};
+export interface IndicesValues {
+  data: typeof FetchIndexApiLogic.values.data;
+  index: ElasticsearchViewIndex | undefined;
+  ingestionMethod: IngestionMethod;
+  ingestionStatus: IngestionStatus;
+  isSyncing: boolean;
+  isWaitingForSync: boolean;
+  lastUpdated: string | null;
+  localSyncNowValue: boolean; // holds local value after update so UI updates correctly
+  syncStatus: SyncStatus;
+}
+
+export const IndexViewLogic = kea<MakeLogicType<IndicesValues, IndicesActions>>({
+  actions: {
+    startSync: true,
+  },
+  connect: {
+    actions: [StartSyncApiLogic, ['makeRequest', 'apiSuccess', 'apiError']],
+    values: [FetchIndexApiLogic, ['data']],
+  },
+  listeners: ({ actions, values }) => ({
+    apiError: (e) => flashAPIErrors(e),
+    apiSuccess: () => {
+      flashSuccessToast(
+        i18n.translate('xpack.enterpriseSearch.content.searchIndex.index.syncSuccess.message', {
+          defaultMessage: 'Successfully scheduled a sync, waiting for a connector to pick it up',
+        })
+      );
+    },
+    makeRequest: () => clearFlashMessages(),
+    startSync: () => actions.makeRequest({ connectorId: values.data?.connector?.id ?? '' }),
+  }),
+  path: ['enterprise_search', 'content', 'indices_logic'],
+  reducers: {
+    localSyncNowValue: [
```

Comment:
I'm a little confused what this is being used for. Optimistic UX while we wait for the makeRequest response?

Reply:
Yeah exactly, I tried to force a makeRequest but Elasticsearch sometimes isn't fast enough in updating and thinks sync_now is still false. This seems like a better solution.